conditionals.md

```diff
@@ -0,0 +1,227 @@
+```
+
+The condition evaluates to false because the value of the numberOfApples is 0. Since 0 is a falsy value, the code block in the else statement will run.
+
+
+### Ternary Operators
+
+Ternary operators is use to simplify an `if...else` statement using short-hand syntax. example:
+
+```js
+let isNightTime = true;
+if(isNightTime) {
+    console.log('Turn on the light');
+} else {
+    console.log('Turn off the light');
+}
+```
+This can be performed using ternary operator:
+
+```js
+let isNightTime = true;
+isNightTime ?  console.log('Turn on the light') : console.log('Turn off the light');
+```
+
+Like if...else statements, ternary operators can be used for conditions which evaluate to true or false.
+
+
+### `else if` statement
+
+`else if` statement is use to add more conditions to our `if...else` to get many possible outcome. The `else if` comes right after the `if` and before the `else` statement. example:
+
+```js
+let stopLight = 'yellow';
+if (stopLight === 'red') {
+    console.log('Stop!');
+} else if (stopLight === 'yellow') {
+    console.log('Slow down');
+} else if (stopLight === 'green') {
+    console.log('Go!');
+}  else {
+    console.log('Caution, unknown');
+}
+```
+
+`if/else if/else` statements are read from top to bottom, so the first condition that evaluates to true from the top to bottom is the block that gets executed.
+
+### Switch Keyword
+
+A `switch` statement provide an alternative way to write the `else if` statement, with a syntax that is easier to read and write. example:
+
+```js
+let stopLight = 'yellow';
+switch (stopLight) {
+    case 'red' :
+      console.log('Stop!');
+      break;
+    case 'yellow' :
+       console.log('Slow down');
+       break;
+    case 'green' :
+       console.log('Go!');
+       break;
+    defualt :
+       console.log('Caution, unknown');
+       break;
+}
+```
+- The switch keyword initiates the statement and is followed by ( ... ), which contains the value that each case will compare. In the example, the value or expression of the switch statement is stopLight.
+- Inside the block, { ... }, there are multiple cases. The case keyword checks if the expression matches the specified value that comes after it. The value following the first case is `'red'`. If the value of stopLight equalled `'red'`, that case‘s console.log() would run.
+- The value of stopLight is `'yellow'`, so the second case runs— `'Slow down'` is logged to the console.
+- The `break` keyword tells the computer to exit the block and not execute any more code or check any other cases inside the code block. Note: Without the `break` keyword at the end of each case, the program would execute the code for all matching cases and the default code as well. This behavior is different from `if/else` conditional statements which execute only one block of code.
+- At the end of each switch statement, there is a default statement. If none of the cases are true, then the code in the default statement will run.
```

Comment:
Few things you should do:
1. Let's add one or two main examples, that illustrate the use of conditionals, this could be a coding challenge from hackerrank, but it should cover at least most of the topic
2. It's very important to add the resources you used, add resources that you used and would recommend readers to look into for more reading.

Reply:
done